subprojects/file-watching/src/main/java/org/gradle/internal/watch/registry/impl/WatchableHierarchies.java

```diff
@@ -46,6 +48,7 @@
     private final Predicate<String> watchFilter;
 
     private FileHierarchySet watchableHierarchies = DefaultFileHierarchySet.of();
+    private FileHierarchySet unwatchableHierarchies = DefaultFileHierarchySet.of();
```

Comment:
I think this should be called something different, like `unsupportedHierarchies`. Using `watchable` vs `unwatchable` would suggest that the two things are in some sort of symmetry, but I don't think that's the case.

Reply:
Adding a bit of Javadoc around these would be nice, too.

Reply:
I renamed and added some Javadocs.